code/modules/scrap/oldificator.dm

```diff
@@ -0,0 +1,250 @@
+/obj/proc/make_old()
+	color = pick("#996633", "#663300", "#666666")
+	light_color = color
+	name = pick("old ", "expired ", "dirty ") + initial(name)
+	desc += pick(" Warranty has expired.", " The inscriptions on this thing were erased by time.", " Looks completely wasted.")
+
+	germ_level = pick(80,110,160)
+	if(prob(40))
+		if(prob(70))
+			light_power = light_power / pick(1.5, 2, 2.5)
+		if(prob(70))
+			light_range = light_range / pick(1.5, 2, 2.5)
+		if(prob(15))
+			light_range = 0
+			light_power = 0
+
+	for(var/obj/item/sub_item in contents)
+		sub_item.make_old()
+
+	/*
+	If anyone ever wishes to make a system
+	of critically broken items, and it's influence on them.
+	Feel free to.
+
+	if(prob(50))
+		crit_fail = 1
+	*/
+
+	update_icon()
+
+/obj/item/make_old()
+	..()
+	if(prob(75))
+		origin_tech = null
+	siemens_coefficient += 0.3
+
+/obj/item/weapon/tool/make_old()
+	..()
+	unreliability += rand(1, 5)
+
+/obj/item/weapon/storage/make_old()
+	var/del_count = rand(0,contents.len)
+	for(var/i = 1 to del_count)
+		var/removed_item = pick(contents)
+		contents -= removed_item
+		qdel(removed_item)
+
+	if(prob(75))
```

Comment:
Add a check for whether it has storage slots at all before doing this.

In general, apply that principle everywhere. Potentially turning zero values into nonzero ones (even negative) can have unintended consequences